hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/security/x509/certificate/client/DefaultCertificateClient.java

```diff
@@ -705,13 +708,28 @@ public synchronized InitResponse init() throws CertificateException {
     if (certificate != null) {
       initCase = initCase | 1;
     }
+
+    Calendar shouldRenewAfter = Calendar.getInstance();
+    shouldRenewAfter
+        .add(Calendar.DAY_OF_YEAR, securityConfig.getRenewalGraceDays());
+    if (initCase == InitCase.ALL.ordinal() &&
```

Comment:
Could we use an enumset for initcase instead of a bitfield? The current solution is very fragile if the InitCase enum ever changes and the code is hard to read as well.

Reply:
Yes we should do this with a more reasonable and readable code, but I would like to defer this to further refactoring, that should come as part of HDDS-7376.

Reply:
Okay that's good. I also like the small refactors you made for now.